Documentation update for v1beta1 api (#632)
kubernetes-sigs/kueue: site/content/en/docs/concepts/cluster_queue.md

A sample ClusterQueue looks like the following:

```yaml
apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:

>   name: "cluster-queue"

Reviewer: any particular reason to add the quotes? In general it's not needed and I prefer to keep things simple.

Reviewer 2: My two cents:
I lean towards not  putting quotation marks around code unless the quotation marks are mandatory part of the code. I suggest not using quotation consistently.

If the K8s SG says otherwise, please disregard my comment.

Author: I couldn't find an official style guide for K8S yamls. I can pass all yamls through some formatter, but would prefer to do it as a separate PR.

Reviewer: I also couldn't find a rule about it, but it looks like most samples don't have quotes.

But doing it in a separate PR sounds good

In a process called [admission](/docs/concepts#admission), Kueue assigns to the
[Workload pod sets](/docs/concepts/workload#pod-sets) a flavor for each resource the pod set
requests.

> Kueue assigns the first flavor in the ClusterQueue's `.spec.resourceGroup.flavors`

Reviewer: ```suggestion
Kueue assigns the first flavor in the ClusterQueue's `.spec.resourceGroups[*].flavors`
```

Author: Done.

Reviewer: Still unresolved. Forgot to push?

Author: Done.

list that has enough unused `nominalQuota` quota in the ClusterQueue or the
ClusterQueue's [cohort](#cohort).

### Resource Groups

It is possible that multiple resources in a ClusterQueue have the same flavors.
This is typical for `cpu` and `memory`, where the flavors are generally tied to

> a machine family or VM availability policies. When two or more resources in a

Reviewer: To tie two or more resources to the same set of flavors, you can list them in the same resource group.

Author: Dome.

ClusterQueue match their flavors, they are said to be a resourceGroup.

> To manage codependent resources, you should list the flavors in the ClusterQueue

Reviewer: Update this paragraph.

Author: I wanted to remove it altogether. Must have missed it. Done.

> An example of a ClusterQueue with codependent resources looks like the following:

Reviewer: ```suggestion
An example of a ClusterQueue with multiple resource groups looks like the following:
```

Author: Done.

```yaml
apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:
  name: "cluster-queue"
spec:
  namespaceSelector: {}
  resourceGroups:
  - coveredResources: ["cpu", "memory"]
    flavors:
    - name: "spot"
      resources:
      - name: "cpu"
        nominalQuota: 9
      - name: "memory"
        nominalQuota: 36Gi
    - name: "on-demand"
      resources:
      - name: "cpu"
        nominalQuota: 18
      - name: "memory"
        nominalQuota: 72Gi
  - coveredResources: ["gpu"]
    flavors:
    - name: "vendor1"
      resources:
      - name: "gpu"
        nominalQuota: 10
    - name: "vendor2"
      resources:
      - name: "gpu"
        nominalQuota: 10
```

> In the example above, `cpu` and `memory` are codependent resources, while `gpu`

Reviewer: ```suggestion
In the example above, `cpu` and `memory` belong to the same resource group and have the same flavors, while `gpu`
```

Reviewer: Still pending

semantics:

- When assigning flavors, Kueue goes through the list of flavors in the
  relevant ResourceGroup inside ClusterQueue's 

>   (`.spec.resourcesGroup.flavors`). For each flavor, Kueue attempts

Reviewer: ```suggestion
  (`.spec.resourcesGroups[*].flavors`). For each flavor, Kueue attempts
```

  to fit a Workload's pod set according to the quota defined in the
  ClusterQueue for the flavor and the unused quota in the cohort.
  If the Workload doesn't fit, Kueue evaluates the next flavor in the list.
- A Workload's pod set resource fits in a flavor defined for a ClusterQueue

>   resourceGroup if the sum of requests for the resource:

Reviewer: I think just resource worked here better.

Reviewer: Still pending

Author: Done.

  `team-a-cq` can admit Workloads with resources adding up to `12+9=21` CPUs and
  `48+36=84Gi` of memory.
- If ClusterQueue `team-b-cq` has pending Workloads and the ClusterQueue
  `team-a-cq` has all its `nominalQuota` quota used, Kueue will admit Workloads in
  ClusterQueue `team-b-cq` before admitting any new Workloads in `team-a-cq`.
  Therefore, Kueue ensures the `nominalQuota` quota for `team-b-cq` is met.

**Note**:

> Kueue supports preemption and Workloads already admitted by Kueue may be 

Reviewer: ```suggestion
Kueue supports [preemption](#preemption) and Workloads already admitted by Kueue may be 
```

Reviewer: Either add the link or just remove the section given that preemption is described below.

Author: Removed.

stopped to make space for new Workloads.

### BorrowingLimit

To limit the amount of resources that a ClusterQueue can borrow from others,

> you can set the `.spec.resourcesGroup[*].flavors[*].resource.borrowingLimit`

Reviewer: ```suggestion
you can set the `.spec.resourcesGroups[*].flavors[*].resources[*].borrowingLimit`
```

Author: Done.

[quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) field.

If, for a given flavor/resource, the `borrowingLimit` field is empty or null, 

> a ClusterQueue can borrow up to the sum of min quotas from all the 

Reviewer: ```suggestion
a ClusterQueue can borrow up to the sum of nominal quotas from all the 
```

Author: Done.
